### PR TITLE
Fix positioning of statically positioned absolute child of inline box

### LINKED ISSUE #29892: Layout 2020: Absolutes with static insets not positioned properly inside inline blocks
Test case:
```html
<div>
    hello
    <div style="display: inline-block; width: 100px; height: 100px; background: blueviolet;">
        <div style="position: absolute; width: 10px; height: 10px; background: cornflowerblue;"></div>
    </div>
    hello
</div>
```

Layout 2020:
image

Firefox:
image

### PR DESCRIPTION
**Pull request by mrobinson**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

### CODE REVIEW

File: components/layout_2020/flow/inline.rs [hunk 1]
```diff
@@ -659,13 +659,20 @@ fn layout_atomic(
                 containing_block_for_children.style.writing_mode,
                 "Mixed writing modes are not supported yet"
             );
-            // FIXME: Do we need to adjust the static position of the hoisted fragments in the positioning
-            // context somewhere near here?
+
+            let collects_for_nearest_positioned_ancestor = ifc
+                .positioning_context
+                .collects_for_nearest_positioned_ancestor();
+            let mut positioning_context =
```
Comment (Loirooriol) [inline components/layout_2020/flow/inline.rs:666]: Nit: `layout_block_level_children_in_parallel` and `layout_block_level_children_sequentially` use the name `child_positioning_context`, which may be slightly clearer than `positioning_context`

**Review by Loirooriol — APPROVED** [review]
Thanks!

File: components/layout_2020/flow/inline.rs (hunk 1 above)
Comment (mrobinson) [reply components/layout_2020/flow/inline.rs]: Good point. I'll make this change.

**Comment by mrobinson** [comment]
@bors-servo r=Loirooriol

Thanks for the review!

**Pull request merged**